### ISSUE TITLE
Web API call to /V1/orders changes Order Increment ID

### ISSUE THREAD

**Issue by Sunil-Nethues** [magento/magento2#17828]
### Preconditions

1. Magento 2.

### Steps to reproduce

1. Create an Order.
2. Note the Order Increment ID
3. Update order via REST API:
`POST http://ce.dev.local/rest/V1/orders`
`Accept: */*`
`Cache-Control: no-cache`
`Content-Type: application/json`
`Authorization: Bearer <API key>`
`{`
  `"entity": {`
   `"entityId": "<Order ID>",`
   `"state": "complete",`
    `"status": "complete"`
  `}`
`}`

### Expected result

1. Order Increment ID remains the same.

### Actual result

1. Order Increment ID is changed.

**Comment by magento-engcom-team** [comment]
Hi @Sunil-Nethues. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@Sunil-Nethues do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by Sunil-Nethues** [comment]
@magento-engcom-team give me 2.2.0 instance

**Comment by magento-engcom-team** [comment]
Hi @Sunil-Nethues. Thank you for your request. I'm working on Magento 2.2.0 instance for you

**Comment by magento-engcom-team** [comment]
Hi @Sunil-Nethues, here is your [Magento instance](https://i-17828-2-2-0.engcom.dev.magento.com).
Admin access: [https://i-17828-2-2-0.engcom.dev.magento.com/admin](https://i-17828-2-2-0.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by Sunil-Nethues** [comment]
thanks @magento-engcom-team can i please request for sample data to reduce time replicating the issue ?

**Comment by ghost** [comment]
@Sunil-Nethues, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. Please provide more detailed steps to reproduce or try to reproduce this issue on a clean installation or latest release.

**Comment by Sunil-Nethues** [comment]
hi @magento-engcom-team i asked for the installation to have sample data installed, i am doing it manually give me some time to replicate it and send you a message.

**Comment by Sunil-Nethues** [comment]
hi @magento-engcom-team we are succeeded in trying to replicate the issue on the fresh instance provided by you as well.

1. I have done following, completed an order using Check/money order
2. created a user with API access
3. Run the shipment creation script from a thrdparty server in this case URL was : https://www.skinoracle.com/shiptest.php?oid=1 {you can use this url too by changing the order id}

I have attached the content of shiptest file with this comment.

[shiptest.zip](https://github.com/magento/magento2/files/2327631/shiptest.zip)

After above url was run the shipment is created but the order increment is changes from 000000001 to 000000002

Please advise further.

**Comment by Sunil-Nethues** [comment]
just an additional information if we pass the increment id while changing the status to complete in above file last call, the order increment id does not changes.

**Comment by Sunil-Nethues** [comment]
hi @magento-engcom-team any information on this ticket please.

Regards
Sunil

**Comment by sunil-gcet** [comment]
hi @magento-engcom-team any information on this ticket please.

Regards
Sunil

**Comment by ghost** [comment]
@Sunil-Nethues, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-pb Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-95038`, `MAGETWO-95039` were created

**Comment by sunil-gcet** [comment]
thanks @magento-engcom-team for acknowledging the issue, looking forward for the fix.

For more information if we send the order increment id with the call above the order increment id is not changed, i thought this might help.

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by milindsingh** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @milindsingh, here is your [Magento instance](https://i-17828-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-17828-2-3-develop.instances.magento-community.engineering/admin](https://i-17828-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by milindsingh** [comment]
@magento-engcom-team @engcom-backlog-pb I am not able to produce the issue regarding increment id change, but I found a new one in Magento 2.3-develop instance.

+ Create shipment with API with following data, and I see a broken Magento Admin UI for Shipment View.
```
{
  "entity": {
    "increment_id": "000000002",
    "order_id": 1,
     "items": [
        {
          "sku": "Test",
          "qty": 1
        }
      ],
    "tracks": [
      {
        "order_id": 1,
        "qty": 1,
        "track_number": "123",
        "title": "Test",
        "carrier_code": "dhl"
      }
    ]  
  }
}
```

![https i 17828 2 3 develop instances magento community engineering admin admin order_shipment view shipment_id 1](https://user-images.githubusercontent.com/13949703/50507095-fbade200-0aa1-11e9-89f0-3b26558f310d.png)

![screenshot from 2018-12-28 13-19-42](https://user-images.githubusercontent.com/13949703/50507392-5bf15380-0aa3-11e9-903a-94b8c3177d99.png)

**Comment by milindsingh** [comment]
#dmcdindia1

**Comment by milindsingh** [comment]
https://github.com/magento-engcom/msi/issues/2220

**Comment by avoelkl** [comment]
I can confirm this Issue still exists in Magento 2.3.2:

Order increment-id before: 69000000009
Order increment-id after the update: 000000077 :x:
`curl -X POST "http://myshop.com/rest/all/V1/orders/" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer 12312123" -d "{ \"entity\": { \"entity_id\": 353, \"extension_attributes\": { \"custom_invoice_number\": \"66666\" } }}"`

It seems in this case Magento picks up a new increment_id from the admin store.

The increment_id needs to be sent via REST as well, then it works as expected:

Order increment-id before: 69000000009
Order increment-id after the update: 69000000009 ✅ 
`curl -X POST "http://myshop.com/rest/all/V1/orders/" -H "accept: application/json" -H "Content-Type: application/json" -H "Authorization: Bearer 12312123" -d "{ \"entity\": { \"entity_id\": 353, \"increment_id\": "69000000009", \"extension_attributes\": { \"custom_invoice_number\": \"66666\" } }}"`

**Comment by magento-engcom-team** [comment]
Hi @Sunil-Nethues, @milindsingh. 

Thank you for your report and collaboration!

The related internal Jira ticket `MAGETWO-95039` was closed as `non-reproducible` in `2.3-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.
